packages/babel-traverse/src/scope/index.js

```diff
@@ -349,9 +349,6 @@ export default class Scope {
     // class expression
     if (local.kind === "local") return;
 
-    // ignore hoisted functions if there's also a local let
-    if (kind === "hoisted" && local.kind === "let") return;
-
```

Comment:
For some reason we have this exception in the code for the case mentioned in the PR. I couldn't come up with a reason why we do this and the commit that introduced it doesn't help either 8a4296a948b496e9fde1d6271a4e0dbdeaacb128
Simply removing it also doesn't break any of the tests. 
Do you think it is safe to remove? Anyone has an idea why this code was there and which case it covers?

Reply:
Maybe it was this? It is an Annex B thing:
```js
let a;
{ function a() {} }
```

Reply:
right but even with removing these lines, this case still works. will add a test.